utils/src/lazy_cache.rs

```diff
@@ -0,0 +1,256 @@
+//! This is a polyfill of the `LazyLock` type in the std library as of Rust 1.80.
```

Comment:
From which version of LazyLock did you add this interface? It does not match the implementation [here](https://doc.rust-lang.org/src/std/sync/lazy_lock.rs.html).

Or is it supposed to be a copy-paste of the interface?

Reply:
It is not meant to be an exact copy-paste because that one was using way too many types which were not accessible from outside rust itself. Meaning, types which were pub(crate). Thus, if I wanted an exact replication I would have had to bring too many lines of code, which I thought we would not need for a simpler feature. It is basically heavily inspired by the implementation, but I removed parts which didn't seem as essential to provide the basic functionality.

Reply:
Makes sense.